odoo/addons/base/models/ir_fields.py

```diff
@@ -263,7 +288,7 @@ def _str_to_selection(self, model, field, value):
         for item, label in selection:
             label = ustr(label)
             labels = [label] + self._get_translations(('selection', 'model', 'code'), label)
-            if value == str(item) or value in labels:
+            if value.lower() == str(item).lower() or any(label for label in labels if value.lower() == label.lower()):
```

Comment:
Why this change exactly?
Also:
`any(value.lower() == label.lower() for label in labels)` does the same thing, right?